### PR TITLE
add processing to the intro overview

### PR DESCRIPTION
**Pull request by tschmetzer**
There were only Python plugins in the intro and I missed a quick introduction note on processing plugin. This is why I chose to write and add this short text passage.

### CODE REVIEW

**Comment by tschmetzer** [comment]
Hm, I don't get it. In line 623 the build says: 
/home/runner/work/QGIS-Documentation/QGIS-Documentation/docs/pyqgis_developer_cookbook/intro.rst:117: WARNING: undefined label: _processing_plugin

But line 2 of processing.rst defines the _processing_plugin:
.. _processing_plugin:

Any hints regarding this are welcome as I don't have a development environment at hand to dig deeper.

**Comment by tschmetzer** [comment]
Content wise I am just wondering why there is mentioning, no reference or manual about scripts that can be run from the processing toolbox. As there is already a good documentation in the manual (see https://docs.qgis.org/3.16/en/docs/user_manual/processing/console.html#creating-scripts-and-running-them-from-the-toolbox) I suggest to make a brief reference to it that points out the distinction between a **Processing Plugin** and a **Processing Script**. Can anybody help me out with this as I don't know the distinction myself?

File: docs/pyqgis_developer_cookbook/intro.rst [hunk 1]
```diff
@@ -111,6 +111,14 @@ for detailed instructions.
 .. index::
   pair: Python; Startup
 
+Processing Plugins
+==================
+
+Processing Plugins can be used to process data. They are easier to develop,
+more specific and more lightweight than Python Plugins. :ref:`_processing_plugin`
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/intro.rst:118]:
```suggestion
more specific and more lightweight than Python Plugins. :ref:`processing_plugin`
```
The leading underscore is used when you set the anchor and removed when you refer to it.

Comment (tschmetzer) [reply docs/pyqgis_developer_cookbook/intro.rst:118]: Oh damn, should've known it. Maybe I should sleep more. Thank you so much!

**Comment by tschmetzer** [comment]
Still failing in line 623:
/home/runner/work/QGIS-Documentation/QGIS-Documentation/docs/pyqgis_developer_cookbook/intro.rst:117: WARNING: Failed to create a cross reference. A title or caption not found: processing_plugin

Could it be the case that `_processing_plugin` is not located in the correct line or context within `processing.rst` or is an anouncement at any index required to that the build process can find it?

**Comment by DelazJ** [comment]
Can you rebase your branch? I think it should be good after 5f5bd54.

**Comment by tschmetzer** [comment]
> 
> 
> Can you rebase your branch? I think it should be good after [5f5bd54](https://github.com/qgis/QGIS-Documentation/commit/5f5bd5406b502696ba74bd6a3d1d7d0037aa8c15).

I merged it and it works now. Thanks for that one!

Any idea who could provide an answer regarding the difference between a Processing Plugin and a Processing Script ? (see above)

File: docs/pyqgis_developer_cookbook/intro.rst [hunk 2]
```diff
@@ -111,6 +111,14 @@ for detailed instructions.
 .. index::
   pair: Python; Startup
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/intro.rst:113]:
```suggestion
```

File: docs/pyqgis_developer_cookbook/intro.rst [hunk 3]
```diff
@@ -111,6 +111,14 @@ for detailed instructions.
...
 
+Processing Plugins
+==================
+
+Processing Plugins can be used to process data. They are easier to develop,
+more specific and more lightweight than Python Plugins. :ref:`processing_plugin`
+explains when the use of Processing algorithms is appropiate and how to develop
+them.
+
 Running Python code when QGIS starts
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/intro.rst:122]: The startup section is actually here
```suggestion
.. index::
  pair: Python; Startup

Running Python code when QGIS starts
```

**Review by DelazJ — COMMENTED** [review]
@tschmetzer I suggest we leave aside details on scripts vs plugins and move forward on this

**Pull request merged**